build.yaml

```diff
@@ -1329,28 +1309,39 @@ steps:
   - kind: runImage
     name: test_python_docs
     image:
-      valueFrom: hail_base_image.image
+      valueFrom: hail_run_image.image
     script: |
       set -ex
-      cd /hail/python/hail
+
+      cp /io/hail.jar $SPARK_HOME/jars/hail.jar
+      cd /io/hail/python/hail
       python3 -m pytest --instafail \
         --doctest-modules \
         --doctest-glob='*.rst' \
         --ignore=docs/conf.py \
         --ignore=docs/doctest_write_data.py
+    inputs:
+      - from: /repo/hail/python/hail
+        to: /io/hail/python/hail
+      - from: /hail.jar
+        to: /io/hail.jar
     dependsOn:
-      - hail_base_image
+      - hail_run_image
+      - build_hail
+      - merge_code
     clouds:
       - gcp
   - kind: runImage
     name: make_docs
     image:
-      valueFrom: hail_base_image.image
+      valueFrom: hail_run_image.image
```

Comment:
Hail is not installed in the `hail_run_image`, I think you can just use hailgenetics/hail here as well if you install the dev dependencies.